perl: Fix regressions in the end-of-life revisions

bernhardkaindl opened the pull request:
- perl@:5.24.1 needs zlib@:1.2.8 - shown by more than a dozen tests
  https://rt.cpan.org/Public/Bug/Display.html?id=120134

- perl@:5.26.2 needs gdbm@:1.14.1 - shown by the test suite
  https://rt-archive.perl.org/perl5/Ticket/Display.html?id=133295

- Fix the test case cpan/Time-Local/t/Local.t to use 4-digit years
  http://blogs.perl.org/users/tom_wyant/2020/01/my-y2020-bug.html

PS: The gitlab CI test of building openssl-1.1.1k fails because 1.1.1k is now deprecated.
I don't know where to update this gitlab test to build the new openssl-1.1.1l instead, or who is best to fix that CI test.

alalazo reviewed (changes requested):
See #25796. Please rebase or push to the same PR instead of opening new ones :slightly_smiling_face:

bernhardkaindl committed:
perl: Fix regressions in the end-of-life revisions

- perl@:5.24.1 needs zlib@:1.2.8 - shown by more than a dozen tests
  https://rt.cpan.org/Public/Bug/Display.html?id=120134

- perl@:5.26.2 needs gdbm@:1.14.1 - shown by the test suite
  https://rt-archive.perl.org/perl5/Ticket/Display.html?id=133295

- Fix the test case cpan/Time-Local/t/Local.t to use 4-digit years
  http://blogs.perl.org/users/tom_wyant/2020/01/my-y2020-bug.html
